rust/connlib/clients/android/src/lib.rs

```diff
@@ -301,7 +304,7 @@ impl Callbacks for CallbackHandler {
 fn throw(env: &mut JNIEnv, class: &str, msg: impl Into<JNIString>) {
     if let Err(err) = env.throw_new(class, msg) {
         // We can't panic, since unwinding across the FFI boundary is UB...
-        tracing::error!("failed to throw Java exception: {err}");
+        tracing::error!(?err, "failed to throw Java exception");
```

Comment:
Doesn't need to be solved in this PR but debug-logging errors doesn't actually give us a whole lot because it only shows the error representation and not the actual messages. I'd say we either display format them or, if it is an anyhow error do it via `:#` which gives you a nice run-down of all causes: https://docs.rs/anyhow/latest/anyhow/struct.Error.html#display-representations

Unfortunately, `tracing` doesn't support the `#` modifier yet (see https://github.com/tokio-rs/tracing/issues/1311) so it would have to look something like:

```suggestion
        tracing::error!("failed to throw Java exception: {:#}", anyhow::Error::new(err));
```

or


```suggestion
        tracing::error!("failed to throw Java exception: {err:#}");
```

if it is already an `anyhow::Error`.

Reply:
Then I'd also like to keep the `err` part, since that would be easier to filter by error type, so what do you think of something like:

```rs
tracing::error!(?err, "failed to throw Java exception: {err:#}");
```
?

Reply:
We could create our own error wrapper:

```rust
struct ErrorWithSources {
	inner: anyhow::Error
}

impl fmt::Display for ErrorWithSources {
	// always forward using alternate modifier
}

fn err_with_sources(err: impl std::error::Error) -> ErrorWithSources { }
```

And then use that as:

```rust
tracing::error!(err = %err_with_sources(err), "failed to throw Java exception");
```